### ISSUE TITLE
details on the code

### ISSUE THREAD

**Issue by anewlearner** [keunwoochoi/residual_block_keras#12]
@keunwoochoi
Thanks for sharing your code. It's really a great help.
I have some questions about the details and I am not sure if my understanding is true. Hope you can help.
- You used `batchnorm->relu->conv`. But it seems to be `conv->batchnorm->relu` in the paper.

```
for i in range(n_skip):
        conv_y = BatchNormalization(axis=1, mode=2)(conv_y)       
        conv_y = Activation('relu')(conv_y)
        if i==0 and is_subsample: # [Subsample at layer 0 if needed]
            conv_y = Convolution2D(n_feature_maps, kernel_row, kernel_col,
                                    subsample=subsample,
                                    border_mode='valid')(conv_y) 
        else:       
            conv_y = Convolution2D(n_feature_maps, kernel_row, kernel_col, border_mode='same')(conv_y)
```
- The description of the code seems to be inconsistent to the code.

> MaxPooling is used instead of strided convolution to make it easier 
>         to set size(output of short-cut) == size(output of conv-layers).
>         If you want to remove MaxPooling,
>            i) change (border_mode in Convolution2D in shortcut), 'same'-->'valid'
>            ii) uncomment ZeroPadding2D in conv layers.
>                (Then the following Conv2D is not the first layer of this container anymore,
>                 so you can remove the input_shape in the line 101, the line with comment #'OPTION' )

Actually, strided convolution is used instead of MaxPooling. 
I changed the code to use MaxPooling.
(i)

```
if is_subsample: # subsample (+ channel expansion if needed)
#    shortcut_y = Convolution2D(n_feature_maps, kernel_sizes[0], kernel_sizes[1],
#                               subsample=subsample,
#                               border_mode='valid')(x)
    shortcut_y = Convolution2D(n_feature_maps, kernel_sizes[0], kernel_sizes[1],
                               border_mode='same')(x)
    shortcut_y = MaxPooling3D(pool_size=(2, 2, 2)))
else: # channel expansion only (e.g. the very first layer of the whole networks)
    if is_expand_channels:
        shortcut_y = Convolution2D(n_feature_maps, 1, 1, border_mode='same')(x)
    else:
    # if no subsample and no channel expension, there's nothing to add on the shortcut.
    shortcut_y = x
```

(ii) comment ZeroPadding2D in conv layers.

**Comment by keunwoochoi** [comment]
> - You used `batchnorm->relu->conv`. But it seems to be `conv->batchnorm->relu` in the paper.

I updated it to follow the author's [updated proposal](https://arxiv.org/abs/1603.05027), where `BN - relu - conv` is used.

> - The description of the code seems to be inconsistent to the code.

You're right. I perhaps would remove the comments. I designed it to use pooling for some reason but these days strided convolution seems quite popular.

To use Max-pooling,  I think you're doing right except `MaxPooling3D`. Shoudn't it be `MaxPooling2D` instead?
